src/tree/updater_quantile_hist.cc

```diff
@@ -480,8 +549,9 @@ void QuantileHistMaker::Builder::Update(const GHistIndexMatrix& gmat,
     p_tree->Stat(nid).base_weight = snode_[nid].weight;
     p_tree->Stat(nid).sum_hess = static_cast<float>(snode_[nid].stats.sum_hess);
   }
-
+  builder_monitor_.Start("PrunerUpdate");
   pruner_->Update(gpair, p_fmat, std::vector<RegTree*>{p_tree});
+  builder_monitor_.Stop("PrunerUpdate");
```

Comment:
Should monitors be a part of Pruner, not Builder?

Reply:
this timer is interesting in context of Builder measure as I understand.

Reply:
Yes, but it's not only one place where the pruner is called

Reply:
this timer was moved